Cannot build PCL with BUILD_apps ON

```
CMakeFiles/pcl_ni_susan.dir/src/ni_susan.cpp.o: dans la fonction « bool pcl::visualization::PCLVisualizer::addPointCloud<pcl::PointXYZRGBL>(pcl::PointCloud<pcl::PointXYZRGBL>::ConstPtr const&, pcl::visualization::PointCloudColorHandler<pcl::PointXYZRGBL> const&, std::string const&, int) »:
ni_susan.cpp:(.text._ZN3pcl13visualization13PCLVisualizer13addPointCloudINS_12PointXYZRGBLEEEbRKNS_10PointCloudIT_E8ConstPtrERKNS0_22PointCloudColorHandlerIS5_EERKSsi[_ZN3pcl13visualization13PCLVisualizer13addPointCloudINS_12PointXYZRGBLEEEbRKNS_10PointCloudIT_E8ConstPtrERKNS0_22PointCloudColorHandlerIS5_EERKSsi]+0x126): référence indéfinie vers « pcl::visualization::PointCloudGeometryHandlerXYZ<pcl::PointXYZRGBL>::PointCloudGeometryHandlerXYZ(boost::shared_ptr<pcl::PointCloud<pcl::PointXYZRGBL> const> const&) »
CMakeFiles/pcl_ni_susan.dir/src/ni_susan.cpp.o:(.rodata._ZTVN3pcl13visualization28PointCloudGeometryHandlerXYZINS_12PointXYZRGBLEEE[_ZTVN3pcl13visualization28PointCloudGeometryHandlerXYZINS_12PointXYZRGBLEEE]+0x30): référence indéfinie vers « pcl::visualization::PointCloudGeometryHandlerXYZ<pcl::PointXYZRGBL>::getGeometry(vtkSmartPointer<vtkPoints>&) const »
collect2: error: ld returned 1 exit status
make[2]: *** [bin/pcl_ni_susan] Erreur 1
make[1]: *** [apps/CMakeFiles/pcl_ni_susan.dir/all] Erreur 2
```

The problem is [here](https://github.com/PointCloudLibrary/pcl/blob/master/apps/src/ni_susan.cpp#L166).

--- a/apps/3d_rec_framework/include/pcl/apps/3d_rec_framework/pipeline/impl/global_nn_recognizer_crh.hpp
+++ b/apps/3d_rec_framework/include/pcl/apps/3d_rec_framework/pipeline/impl/global_nn_recognizer_crh.hpp
@@ -336,7 +336,7 @@ template<template<class > class Distance, typename PointInT, typename FeatureT>
           pcl::transformPointCloud (*model_cloud, *model_aligned, transforms_->at (i));
 
           pcl::IterativeClosestPoint<PointInT, PointInT> reg;
-          reg.setInputCloud (model_aligned); //model
+          reg.setInputSource (model_aligned); //model
           reg.setInputTarget (cloud_voxelized_icp); //scene
           reg.setMaximumIterations (ICP_iterations_);
           reg.setMaxCorrespondenceDistance (VOXEL_SIZE_ICP_ * 3.f);

--- a/apps/cloud_composer/include/pcl/apps/cloud_composer/cloud_view.h
+++ b/apps/cloud_composer/include/pcl/apps/cloud_composer/cloud_view.h
@@ -152,3 +152,4 @@ namespace pcl
 
 Q_DECLARE_METATYPE (pcl::cloud_composer::CloudView);
 #endif
+

--- a/apps/cloud_composer/include/pcl/apps/cloud_composer/project_model.h
+++ b/apps/cloud_composer/include/pcl/apps/cloud_composer/project_model.h
@@ -148,7 +148,7 @@ namespace pcl
         
         /** \brief Slot Called whenever the item selection_model_ changes */
         void
-        itemSelectionChanged ( const QItemSelection & selected, const QItemSelection & deselected );
+        itemSelectionChanged ( const QItemSelection &, const QItemSelection &);
         
         /** \brief Creates a new cloud from the selected items and points */
         void 

--- a/apps/cloud_composer/src/cloud_view.cpp
+++ b/apps/cloud_composer/src/cloud_view.cpp
@@ -195,7 +195,7 @@ pcl::cloud_composer::CloudView::selectedItemChanged (const QItemSelection & sele
 }
 
 void
-pcl::cloud_composer::CloudView::dataChanged (const QModelIndex & topLeft, const QModelIndex & bottomRight)
+pcl::cloud_composer::CloudView::dataChanged (const QModelIndex &, const QModelIndex &)
 {
     
   
@@ -283,7 +283,7 @@ pcl::cloud_composer::CloudView::setInteractorStyle (interactor_styles::INTERACTO
 }
 
 void
-pcl::cloud_composer::CloudView::selectionCompleted (vtkObject* caller, unsigned long event_id, void* client_data, void* call_data)
+pcl::cloud_composer::CloudView::selectionCompleted (vtkObject*, unsigned long, void*, void* call_data)
 {
   boost::shared_ptr<SelectionEvent> selected (static_cast<SelectionEvent*> (call_data));
   
@@ -297,7 +297,7 @@ pcl::cloud_composer::CloudView::selectionCompleted (vtkObject* caller, unsigned 
 
 
 void
-pcl::cloud_composer::CloudView::manipulationCompleted (vtkObject* caller, unsigned long event_id, void* client_data, void* call_data)
+pcl::cloud_composer::CloudView::manipulationCompleted (vtkObject*, unsigned long, void*, void* call_data)
 {
   boost::shared_ptr<ManipulationEvent> manip_event (static_cast<ManipulationEvent*> (call_data));
   
@@ -308,3 +308,4 @@ pcl::cloud_composer::CloudView::manipulationCompleted (vtkObject* caller, unsign
    
   }
 }
+

--- a/apps/cloud_composer/src/commands.cpp
+++ b/apps/cloud_composer/src/commands.cpp
@@ -475,12 +475,11 @@ pcl::cloud_composer::DeleteItemCommand::DeleteItemCommand (QList <const CloudCom
 }
 
 bool
-pcl::cloud_composer::DeleteItemCommand::runCommand (AbstractTool* tool)
+pcl::cloud_composer::DeleteItemCommand::runCommand (AbstractTool*)
 {
   
   //For delete item command, each selected item should be processed separately
   //e.g. delete every selected item
-  int num_new_items = 0;
   foreach (const CloudComposerItem *item, original_data_)
   {
     QList <CloudComposerItem*> output;
@@ -593,3 +592,4 @@ pcl::cloud_composer::MergeCloudCommand::redo ()
   
   }
 }
+

--- a/apps/cloud_composer/src/item_inspector.cpp
+++ b/apps/cloud_composer/src/item_inspector.cpp
@@ -47,7 +47,7 @@ pcl::cloud_composer::ItemInspector::setModel (ProjectModel* new_model)
 }
 
 void
-pcl::cloud_composer::ItemInspector::selectionChanged (const QModelIndex &current, const QModelIndex &)
+pcl::cloud_composer::ItemInspector::selectionChanged (const QModelIndex &, const QModelIndex &)
 {
   //If we have a model loaded, save its tree state 
  // if (current_item_properties_model_)

--- a/apps/cloud_composer/src/items/cloud_composer_item.cpp
+++ b/apps/cloud_composer/src/items/cloud_composer_item.cpp
@@ -60,13 +60,13 @@ pcl::cloud_composer::CloudComposerItem::addChild (CloudComposerItem *item_arg)
 }
 
 void
-pcl::cloud_composer::CloudComposerItem::paintView (boost::shared_ptr<pcl::visualization::PCLVisualizer> vis) const
+pcl::cloud_composer::CloudComposerItem::paintView (boost::shared_ptr<pcl::visualization::PCLVisualizer>) const
 {
   qDebug () << "Paint View in Cloud Composer Item - doing nothing";
 }
 
 void
-pcl::cloud_composer::CloudComposerItem::removeFromView (boost::shared_ptr<pcl::visualization::PCLVisualizer> vis) const
+pcl::cloud_composer::CloudComposerItem::removeFromView (boost::shared_ptr<pcl::visualization::PCLVisualizer>) const
 {
   qDebug () << "Remove from View in Cloud Composer Item - doing nothing";
 }

--- a/apps/cloud_composer/src/project_model.cpp
+++ b/apps/cloud_composer/src/project_model.cpp
@@ -380,7 +380,7 @@ pcl::cloud_composer::ProjectModel::saveSelectedCloudToFile ()
   pcl::PCLPointCloud2::ConstPtr cloud = cloud_to_save->data (ItemDataRole::CLOUD_BLOB).value <pcl::PCLPointCloud2::ConstPtr> ();
   Eigen::Vector4f origin = cloud_to_save->data (ItemDataRole::ORIGIN).value <Eigen::Vector4f> ();
   Eigen::Quaternionf orientation = cloud_to_save->data (ItemDataRole::ORIENTATION).value <Eigen::Quaternionf> ();
-  int result = pcl::io::savePCDFile (filename.toStdString (), *cloud, origin, orientation );
+  pcl::io::savePCDFile (filename.toStdString (), *cloud, origin, orientation );
   
 }
 
@@ -539,7 +539,6 @@ void
 pcl::cloud_composer::ProjectModel::selectAllItems (QStandardItem* item)
 {
  
-  int num_rows;
   if (!item)
     item = this->invisibleRootItem ();
   else
@@ -564,7 +563,6 @@ pcl::cloud_composer::ProjectModel::emitAllStateSignals ()
   emit newCloudFromSelectionAvailable (onlyCloudItemsSelected ());  
   
   //Find out which style is active, emit the signal 
-  QMap<interactor_styles::INTERACTOR_STYLES, bool>::iterator itr = selected_style_map_.begin();
   foreach (interactor_styles::INTERACTOR_STYLES style, selected_style_map_.keys())
   {
     if (selected_style_map_.value (style))
@@ -578,7 +576,7 @@ pcl::cloud_composer::ProjectModel::emitAllStateSignals ()
 }
 
 void
-pcl::cloud_composer::ProjectModel::itemSelectionChanged ( const QItemSelection & selected, const QItemSelection & deselected )
+pcl::cloud_composer::ProjectModel::itemSelectionChanged ( const QItemSelection &, const QItemSelection &)
 {
   //qDebug () << "Item selection changed!";
   //Set all point selected cloud items back to green text, since if they are selected they get changed to white

--- a/apps/cloud_composer/src/properties_model.cpp
+++ b/apps/cloud_composer/src/properties_model.cpp
@@ -128,9 +128,10 @@ pcl::cloud_composer::PropertiesModel::copyProperties (const PropertiesModel* to_
 
 
 void
-pcl::cloud_composer::PropertiesModel::propertyChanged (QStandardItem* property_item)
+pcl::cloud_composer::PropertiesModel::propertyChanged (QStandardItem*)
 {
   //qDebug () << "Property Changed in properties model";
   parent_item_->propertyChanged ();
   
 }
+

--- a/apps/cloud_composer/src/toolbox_model.cpp
+++ b/apps/cloud_composer/src/toolbox_model.cpp
@@ -73,7 +73,7 @@ pcl::cloud_composer::ToolBoxModel::addToolGroup (QString tool_group_name)
 }
 
 void 
-pcl::cloud_composer::ToolBoxModel::activeProjectChanged(ProjectModel* new_model, ProjectModel* previous_model)
+pcl::cloud_composer::ToolBoxModel::activeProjectChanged(ProjectModel* new_model, ProjectModel*)
 {
   //Disconnect old project model signal for selection change
   if (project_model_)
@@ -132,7 +132,7 @@ pcl::cloud_composer::ToolBoxModel::toolAction ()
 }
 
 void 
-pcl::cloud_composer::ToolBoxModel::selectedItemChanged ( const QItemSelection & selected, const QItemSelection & deselected )
+pcl::cloud_composer::ToolBoxModel::selectedItemChanged ( const QItemSelection & selected, const QItemSelection &)
 {
   updateEnabledTools (selected);
 }

--- a/apps/cloud_composer/tools/sanitize_cloud.cpp
+++ b/apps/cloud_composer/tools/sanitize_cloud.cpp
@@ -20,7 +20,7 @@ pcl::cloud_composer::SanitizeCloudTool::~SanitizeCloudTool ()
 }
 
 QList <pcl::cloud_composer::CloudComposerItem*>
-pcl::cloud_composer::SanitizeCloudTool::performAction (ConstItemList input_data, PointTypeFlags::PointType type)
+pcl::cloud_composer::SanitizeCloudTool::performAction (ConstItemList input_data, PointTypeFlags::PointType)
 {
   QList <CloudComposerItem*> output;
   const CloudComposerItem* input_item;
@@ -81,3 +81,4 @@ pcl::cloud_composer::SanitizeCloudToolFactory::createToolParameterModel (QObject
   
   return parameter_model;
 }
+

--- a/apps/cloud_composer/tools/voxel_grid_downsample.cpp
+++ b/apps/cloud_composer/tools/voxel_grid_downsample.cpp
@@ -22,7 +22,7 @@ pcl::cloud_composer::VoxelGridDownsampleTool::~VoxelGridDownsampleTool ()
 }
 
 QList <pcl::cloud_composer::CloudComposerItem*>
-pcl::cloud_composer::VoxelGridDownsampleTool::performAction (ConstItemList input_data, PointTypeFlags::PointType type)
+pcl::cloud_composer::VoxelGridDownsampleTool::performAction (ConstItemList input_data, PointTypeFlags::PointType)
 {
   QList <CloudComposerItem*> output;
   const CloudComposerItem* input_item;

--- a/apps/modeler/src/icp_registration_worker.cpp
+++ b/apps/modeler/src/icp_registration_worker.cpp
@@ -138,7 +138,7 @@ pcl::modeler::ICPRegistrationWorker::processImpl(CloudMeshItem* cloud_mesh_item)
   // Set the euclidean distance difference epsilon (criterion 3)
   icp.setEuclideanFitnessEpsilon (*euclidean_fitness_epsilon_);
 
-  icp.setInputCloud(cloud_mesh_item->getCloudMesh()->getCloud());
+  icp.setInputSource(cloud_mesh_item->getCloudMesh()->getCloud());
   icp.setInputTarget(cloud_);
   pcl::PointCloud<CloudMesh::PointT> result;
   icp.align(result);

--- a/apps/point_cloud_editor/include/pcl/apps/point_cloud_editor/cutCommand.h
+++ b/apps/point_cloud_editor/include/pcl/apps/point_cloud_editor/cutCommand.h
@@ -93,13 +93,6 @@ class CutCommand : public Command
       assert(false); return (*this);
     }
 
-    /// The copy buffer which backs up the points removed from the cloud.
-    CopyBuffer cut_cloud_buffer_;
-
-    /// a selection which backs up the index of the points cut in the
-    /// original cloud.
-    Selection cut_selection_;
-
     /// A shared pointer pointing to the selection object.
     SelectionPtr selection_ptr_;
 
@@ -108,6 +101,13 @@ class CutCommand : public Command
 
     /// a pointer pointing to the copy buffer.
     CopyBufferPtr copy_buffer_ptr_;
+
+    /// a selection which backs up the index of the points cut in the
+    /// original cloud.
+    Selection cut_selection_;
+
+    /// The copy buffer which backs up the points removed from the cloud.
+    CopyBuffer cut_cloud_buffer_;
 
 };
 

--- a/apps/point_cloud_editor/include/pcl/apps/point_cloud_editor/deleteCommand.h
+++ b/apps/point_cloud_editor/include/pcl/apps/point_cloud_editor/deleteCommand.h
@@ -89,18 +89,18 @@ class DeleteCommand : public Command
       assert(false); return (*this);
     }
 
-    /// a copy buffer which backs up the points deleted from the cloud.
-    CopyBuffer deleted_cloud_buffer_;
+    /// a pointer pointing to the cloud
+    CloudPtr cloud_ptr_;
+
+    /// A shared pointer pointing to the selection object.
+    SelectionPtr selection_ptr_;
 
     /// a selection which backs up the index of the deleted points in the
     /// original cloud.
     Selection deleted_selection_;
 
-    /// A shared pointer pointing to the selection object.
-    SelectionPtr selection_ptr_;
-
-    /// a pointer pointing to the cloud
-    CloudPtr cloud_ptr_;
+    /// a copy buffer which backs up the points deleted from the cloud.
+    CopyBuffer deleted_cloud_buffer_;
 };
 
 #endif // DELETE_COMMAND_H_

--- a/apps/point_cloud_editor/include/pcl/apps/point_cloud_editor/pasteCommand.h
+++ b/apps/point_cloud_editor/include/pcl/apps/point_cloud_editor/pasteCommand.h
@@ -92,10 +92,8 @@ class PasteCommand : public Command
       assert(false); return (*this);
     }
 
-    /// The size of the cloud before new points are pasted. This value is used
-    /// to mark the point where points were added to the cloud. In order to
-    /// support undo, one only has to resize the cloud using this value.
-    unsigned int prev_cloud_size_;
+    /// a pointer pointing to the copy buffer.
+    ConstCopyBufferPtr copy_buffer_ptr_;
 
     /// A shared pointer pointing to the selection object.
     SelectionPtr selection_ptr_;
@@ -103,7 +101,9 @@ class PasteCommand : public Command
     /// a pointer pointing to the cloud
     CloudPtr cloud_ptr_;
 
-    /// a pointer pointing to the copy buffer.
-    ConstCopyBufferPtr copy_buffer_ptr_;
+    /// The size of the cloud before new points are pasted. This value is used
+    /// to mark the point where points were added to the cloud. In order to
+    /// support undo, one only has to resize the cloud using this value.
+    unsigned int prev_cloud_size_;
 };
 #endif // PASTE_COMMAND_H_

--- a/apps/point_cloud_editor/include/pcl/apps/point_cloud_editor/transformCommand.h
+++ b/apps/point_cloud_editor/include/pcl/apps/point_cloud_editor/transformCommand.h
@@ -92,20 +92,20 @@ class TransformCommand : public Command
     void
     applyTransform(ConstSelectionPtr sel_ptr);
 
+    /// pointers to constructor params
+    ConstSelectionPtr selection_ptr_;
+
+    /// a pointer poiting to the cloud
+    CloudPtr cloud_ptr_;
+
+    float translate_x_, translate_y_, translate_z_;
+
     /// An internal selection object used to perform undo
     SelectionPtr internal_selection_ptr_;
 
     /// the transform matrix to be used to compute the new coordinates
     /// of the selected points
     float transform_matrix_[MATRIX_SIZE];
-
-    float translate_x_, translate_y_, translate_z_;
-
-    /// pointers to constructor params
-    ConstSelectionPtr selection_ptr_;
-
-    /// a pointer poiting to the cloud
-    CloudPtr cloud_ptr_;
 
     /// The transform matrix of the cloud used by this command
     float cloud_matrix_[MATRIX_SIZE];

--- a/apps/point_cloud_editor/src/statisticsDialog.cpp
+++ b/apps/point_cloud_editor/src/statisticsDialog.cpp
@@ -40,7 +40,7 @@
 
 #include <pcl/apps/point_cloud_editor/statisticsDialog.h>
 
-StatisticsDialog::StatisticsDialog(QWidget *parent)
+StatisticsDialog::StatisticsDialog(QWidget *)
 {
   button_box_ = new QDialogButtonBox;
   button_box_->addButton(tr("Hide"), QDialogButtonBox::AcceptRole);

--- a/apps/src/openni_klt.cpp
+++ b/apps/src/openni_klt.cpp
@@ -219,7 +219,7 @@ class OpenNIViewer
 
       tracker_.reset (new pcl::tracking::PyramidalKLTTracker<PointType>);
 
-      bool image_init = false, cloud_init = false;
+      bool image_init = false;
 
       grabber_.start ();
 

--- a/keypoints/include/pcl/keypoints/impl/harris_2d.hpp
+++ b/keypoints/include/pcl/keypoints/impl/harris_2d.hpp
@@ -105,7 +105,6 @@ pcl::HarrisKeypoint2D<PointInT, PointOutT, IntensityT>::computeSecondMomentMatri
   
   int x = static_cast<int> (index % input_->width);
   int y = static_cast<int> (index / input_->width);
-  unsigned count = 0;
   // indices        0   1   2
   // coefficients: ixix  ixiy  iyiy
   memset (coefficients, 0, sizeof (float) * 3);

--- a/keypoints/include/pcl/keypoints/impl/trajkovic_2d.hpp
+++ b/keypoints/include/pcl/keypoints/impl/trajkovic_2d.hpp
@@ -216,7 +216,6 @@ pcl::TrajkovicKeypoint2D<PointInT, PointOutT, IntensityT>::detectKeypoints (Poin
   std::vector<bool> occupency_map (indices.size (), false);
   const int width (input_->width);
   const int height (input_->height);
-  const int occupency_map_size (indices.size ());
 
 #ifdef _OPENMP
 #pragma omp parallel for shared (output) num_threads (threads_)

--- a/keypoints/include/pcl/keypoints/impl/trajkovic_3d.hpp
+++ b/keypoints/include/pcl/keypoints/impl/trajkovic_3d.hpp
@@ -230,7 +230,6 @@ pcl::TrajkovicKeypoint3D<PointInT, PointOutT, NormalT>::detectKeypoints (PointCl
   std::vector<bool> occupency_map (indices.size (), false);
   const int width (input_->width);
   const int height (input_->height);
-  const int occupency_map_size (indices.size ());
 
 #ifdef _OPENMP
 #pragma omp parallel for shared (output) num_threads (threads_)

--- a/visualization/src/point_cloud_handlers.cpp
+++ b/visualization/src/point_cloud_handlers.cpp
@@ -702,7 +702,7 @@ pcl::visualization::PointCloudGeometryHandlerCustom<pcl::PCLPointCloud2>::PointC
 
 // Instantiations of specific point types
 #ifdef PCL_ONLY_CORE_POINT_TYPES
-  PCL_INSTANTIATE(PointCloudGeometryHandlerXYZ, (pcl::PointSurfel)(pcl::PointXYZ)(pcl::PointXYZI)(pcl::PointXYZRGB)(pcl::PointXYZRGBA)(pcl::PointNormal)(pcl::PointXYZRGBNormal))
+  PCL_INSTANTIATE(PointCloudGeometryHandlerXYZ, (pcl::PointSurfel)(pcl::PointXYZ)(pcl::PointXYZI)(pcl::PointXYZRGB)(pcl::PointXYZRGBA)(pcl::PointNormal)(pcl::PointXYZRGBNormal)(pcl::PointXYZRGBL))
   PCL_INSTANTIATE(PointCloudGeometryHandlerSurfaceNormal, (pcl::Normal)(pcl::PointNormal)(pcl::PointXYZRGBNormal))
 #else
   PCL_INSTANTIATE(PointCloudGeometryHandlerXYZ, PCL_XYZ_POINT_TYPES)

